includes/Assets.php

```diff
@@ -144,6 +152,38 @@ public function register_script_asset( string $script_handle, array $script_depe
 		);
 
 		wp_set_script_translations( $script_handle, 'web-stories' );
+
+		// Register every dynamically imported chunk as a script, just so
+		// that we can print their translations whenever the main script is enqueued.
+		// The actual enqueueing of these chunks is done by the main script via dynamic imports.
+		foreach ( $asset['chunks'] as $dynamic_chunk ) {
+			$this->register_script(
+				$dynamic_chunk,
+				$base_script_path . $dynamic_chunk . '.js',
+				[],
+				WEBSTORIES_VERSION,
+				$in_footer
+			);
+
+			wp_set_script_translations( $dynamic_chunk, 'web-stories' );
+
+			$json_translations = load_script_textdomain( $dynamic_chunk, 'web-stories', null );
+
+			if ( ! $json_translations ) {
+				// Register empty locale data object to ensure the domain still exists.
+				$json_translations = '{ "locale_data": { "messages": { "": {} } } }';
+			}
+
+			$output = <<<JS
+( function( domain, translations ) {
+	var localeData = translations.locale_data[ domain ] || translations.locale_data.messages;
+	localeData[""].domain = domain;
+	wp.i18n.setLocaleData( localeData, domain );
+} )( "web-stories", {$json_translations} );
+JS;
+
+			wp_add_inline_script( $script_handle, $output );
```

Comment:
Could / should we break this crunch into its own method?

Reply:
Can't we just call [print_translations](https://github.com/WordPress/wordpress-develop/blob/4205214eba6a467b2dd97e866327e3d17a5c6a0a/src/wp-includes/class.wp-scripts.php#L591) here?

Reply:
> Can't we just call [print_translations](https://github.com/WordPress/wordpress-develop/blob/4205214eba6a467b2dd97e866327e3d17a5c6a0a/src/wp-includes/class.wp-scripts.php#L591) here?

No. `print_translations()` returns a `<script>` tag. `wp_add_inline_script()` does not allow `<script>` tags. It requires the plain JS.

Reply:
> No. print_translations() returns a <script> tag. wp_add_inline_script() does not allow <script> tags. 

If you pass $echo = false, it doesn't print / wrap in a script tag.

Reply:
Fixed now